### PR TITLE
Fix for optional route def properties when `exactOptionalPropertyTypes` is enabled downstream

### PR DESCRIPTION
**Pull request by codetheweb**

### CODE REVIEW

File: src/route-def.ts [hunk 1]
```diff
@@ -3,12 +3,12 @@ import { HTTPMethod } from "./http-method"
 export type RouteDef = {
   route: string
   method: HTTPMethod
-  queryParams?: Record<string, any>
-  jsonBody?: Record<string, any>
-  commonParams?: Record<string, any>
-  formData?: Record<string, any>
-  jsonResponse?: Record<string, any>
-  urlEncodedFormData?: Record<string, any>
```
Comment (abimaelmartell) [inline src/route-def.ts:11]: why is this necessary, i believe we had something like this on another project and just used the `exactOptionalPropertyTypes` option on tsconfig

Comment (codetheweb) [reply src/route-def.ts:11]: yeah, but `exactOptionalPropertyTypes` interacts weirdly with dependencies (at least I wouldn't expect it to be applied for dependencies)


but in fake-schlage with `exactOptionalPropertyTypes: true` enabled in `fake-schlage`:
Screenshot 2024-02-29 at 15 04 24

after this change the error in fake-schlage goes away

Comment (abimaelmartell) [reply src/route-def.ts:11]: yeah i think makes sense to be flexible here

**Review by abimaelmartell — APPROVED** [review]

File: src/route-def.ts (hunk 1 above)
Comment (abimaelmartell) [reply src/route-def.ts:11]: that setting would need to be changed on project that depend on this
